localstack/config.py

```diff
@@ -411,12 +408,113 @@ def in_docker():
 # whether to forward edge requests in-memory (instead of via proxy servers listening on backend ports)
 # TODO: this will likely become the default and may get removed in the future
 FORWARD_EDGE_INMEM = True
-# Default bind address for the edge service
-EDGE_BIND_HOST = os.environ.get("EDGE_BIND_HOST", "").strip() or "127.0.0.1"
-# port number for the edge service, the main entry point for all API invocations
-EDGE_PORT = int(os.environ.get("EDGE_PORT") or 0) or DEFAULT_PORT_EDGE
-# fallback port for non-SSL HTTP edge service (in case HTTPS edge service cannot be used)
-EDGE_PORT_HTTP = int(os.environ.get("EDGE_PORT_HTTP") or 0)
+
+# expose services on a specific host externally
+# DEPRECATED:  since v2.0.0 as we are moving to LOCALSTACK_HOST
+HOSTNAME_EXTERNAL = os.environ.get("HOSTNAME_EXTERNAL", "").strip() or LOCALHOST
+
+# name of the host under which the LocalStack services are available
+# DEPRECATED: if the user sets this since v2.0.0 as we are moving to LOCALSTACK_HOST
+LOCALSTACK_HOSTNAME = os.environ.get("LOCALSTACK_HOSTNAME", "").strip() or LOCALHOST
+
+
+def populate_legacy_edge_configuration(
+    environment: Dict[str, str]
+) -> Tuple[str, str, str, int, int]:
+    if is_in_docker:
+        default_ip = "0.0.0.0"
+    else:
+        default_ip = "127.0.0.1"
+
+    localstack_host_raw = environment.get("LOCALSTACK_HOST")
+    edge_bind_raw = environment.get("GATEWAY_LISTEN")
+
+    # new for v2
+    # populate LOCALSTACK_HOST first since GATEWAY_LISTEN may be derived from LOCALSTACK_HOST
+    localstack_host = localstack_host_raw
+    if localstack_host is None:
+        localstack_host = f"{constants.LOCALHOST_HOSTNAME}:{constants.DEFAULT_PORT_EDGE}"
+
+    def parse_edge_bind(value: str) -> str:
+        if ":" in value:
+            ip, port_s = value.split(":", 1)
+            if not ip.strip():
+                ip = default_ip
+            if not port_s.strip():
+                port_s = "4566"
+            port = int(port_s)
+            return f"{ip}:{port}"
+        else:
+            return f"{value}:4566"
+
+    edge_bind = edge_bind_raw
+    if edge_bind is None:
+        # default to existing behaviour
+        port = int(localstack_host.split(":")[-1])
+        edge_bind = f"{default_ip}:{port}"
+    else:
+        components = edge_bind.split(",")
```

Comment:
nit: should we maybe output a warning here if there are >1 components in here for now, since they are ignored? we could however pretty soon update the logic to start the gateway and proxies.